### PR TITLE
feat : added sortByRating, added clear filters

### PR DESCRIPTION
**Pull request by Sharmaryan**
- added filter by price range functionality
- added sort by rating functionality
- added clear filter functionality
- not responsive

### CODE REVIEW

File: src/components/Products/Products.jsx [hunk 1]
```diff
@@ -5,6 +5,8 @@ import { useProduct } from "../../context/product-context";
 import { filterByCategory } from "../../SortingAndFiltering/filterByCategory";
 
 import { sortByPrice } from "../../SortingAndFiltering/sortByPrice";
+import { sortByRating } from "../../SortingAndFiltering/sortByRating";
+import {filterByPriceRange} from '../../SortingAndFiltering/filterByPriceRange';
```
Comment (anuj-kumary) [inline src/components/Products/Products.jsx]: Suggestion: can use barrel export and import directly from `./sortingAndFiltering`

**Review by anuj-kumary — APPROVED** [review]
Loved how you set up your folder structure
Clean code gave better readability
Good naming conventions as well

File: src/components/Products/Products.jsx [hunk 2]
```diff
@@ -36,11 +39,14 @@ export const Products = () => {
   );
   const sortedProducts = sortByPrice(sortBy, filteredProducts);
 
+  const ratedProducts = sortByRating(sortedProducts, rating);
```
Comment (sakshi006) [inline src/components/Products/Products.jsx:38]: Here you can use the concept of composition of functions and use it here
For example: [see this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#function_composition_enabling_piping)

**Review by sakshi006 — APPROVED** [review]
The code structure is good, its readable.
Naming convention conveys what that particular element might be used for.
One suggestion : For sidebar filters, add an hover effect, that would look better.
Also, you can use an index file for exporting the utility functions.

I liked the how you created a separate component for rating and then mapped through it for its render.

File: src/components/Products/Products.jsx (hunk 1 above)
Comment (Sharmaryan) [reply src/components/Products/Products.jsx]: done

**Pull request merged**